class-auth.php

```diff
@@ -351,14 +355,15 @@ public function validate_token( $return_response = true ) {
 		list($token) = sscanf( $auth, 'Bearer %s' );
 
 		if (!$token) {
-            // Check if Basic Auth is used and create a token if so
-            list($username, $password) = explode( ':', base64_decode( substr( $auth, 6 ) ) );
-            $request = new WP_REST_Request( 'POST', '/wp-json/jwt-auth/v1/token' );
-            $request->set_param( 'username', $username );
-            $request->set_param( 'password', $password );
-            $token = $this->generate_token( $request );
-            if (is_array($token) && isset($token['token'])) $token = $token['token'];
-            return;
+            return new WP_REST_RESPONSE(
```

Comment:
This is sent if the token **doens't ** exist?

Reply:
Cool, might make sense to add a comment that we are accepting the basic auth or something like that.

Another pattern that helps with readability is single return  per function _(combined with fail fast or exceptional conditions evaluated first)_, this especially helps in statically typed languages or if there is a linter because those things can pick up any path that doesn't return.

https://coderanch.com/t/685748/java/Single-return

Reply:
I think some of the issue with single return per function here is that the repo I forked it from has a lot of validation that is going to fail after this if statement because it doesn't have a jwt token. I will make it more readable with comments but I think I'd need to rewrite a few things to make single return workable for this function.